### PR TITLE
Partial thread per core, again

### PR DESCRIPTION
**Pull request by rklaehn**

### CODE REVIEW

**Comment by rklaehn** [comment]
https://github.com/n0-computer/iroh/pull/1088 was not against main, but against x-hp-tpc-2 due to the whole x-hp confusion last week. This is identical.

**Review by dignifiedquire — APPROVED** [review]

File: src/runtime.rs [hunk 1]
```diff
@@ -0,0 +1,420 @@
+//! The runtime module provides the iroh runtime, consisting of a general purpose
+//! tokio runtime and a set of single threaded runtimes.
+//!
+//! Functions spawned on single threaded runtimes should not panic,
+//! because panics will shut down the worker thread running the function, and
+//! the runtime will not spawn new threads.
+//!
+//! It is best to run the entire program with panic = 'abort'.
```
Comment (flub) [inline src/runtime.rs:8]: should we put this in Cargo.toml?

Comment (Arqu) [reply src/runtime.rs:8]: The release builds already do that

Comment (flub) [reply src/runtime.rs:8]: great, thanks!

**Pull request merged**